client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -75,6 +76,7 @@
   private static final String SPARK_HOME_KEY = "spark.home";
   private static final String DRIVER_OPTS_KEY = "spark.driver.extraJavaOptions";
   private static final String EXECUTOR_OPTS_KEY = "spark.executor.extraJavaOptions";
+  private static final SparkLauncher launcher = new SparkLauncher();
```

Comment:
Again: this should not be a class field. It should be a local variable in the only method where it is used.